### PR TITLE
fix openff pin, and add upper pin for openff and openmmtools

### PR DESCRIPTION
**Pull request by mikemhenry**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mikemhenry** [comment]
Okay this is ready to get merged in:
https://github.com/conda-forge/openfe-feedstock/pull/35
tl;dr
* drop `pip check` and run all our unit tests (we were not running `openfecli`) using xdist
* set git_depth to -1 so we don't need to fuss with it (clones whole repo, it is fast)
* sets the pins to `openff-toolkit >=0.13,<0.14` and `openmmtools >=0.23.0,<0.24` to avoid breakage

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,10 +29,10 @@ requirements:
     - lomap2 >=2.1.0
     - networkx
     - numpy <1.24
-    - openff-toolkit ==0.13
+    - openff-toolkit >=0.13,<0.14
```
Comment (richardjgowers) [inline recipe/meta.yaml:32]: Isn't this the same? I thought ==0.13 would allow any 0.13.x release?

Comment (mikemhenry) [reply recipe/meta.yaml:32]: Almost! See https://conda.io/projects/conda-build/en/latest/resources/package-spec.html#examples

If you want a fuzzy match like that, you need only one `=`, as written that will match `0.13` or `0.13.0` or `0.13.0.0` etc

I dislike that syntax, it is a huge foot gun, so I like the explicitness of a range. I could change it to `openff-toolkit =0.13` if you want

**Review by richardjgowers — APPROVED** [review]
